### PR TITLE
feat(tai): add app

### PR DESCRIPTION
**Pull request by tonzgao**
## Description

https://discord.com/channels/647279669388771329/964569348243075082/1113521354810069043

note: removed position presenter and saviour from tai implementation (reflexer fork). this pr is only the base vaults. also does not include the bunni weth/tai lp, since im not sure how long that will exist for and also requires bunni zapper implementation

hacky: currently collateral types are pulled from github json file. cant find equivalent in the gql or any contract.

## Checklist

- [X] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [X] (optional) As a contributor, my Ethereum address/ENS is: tonygao.eth

## How to test?

Random address: http://localhost:5001/apps/tai/balances?network=ethereum&addresses%5B%5D=0x20eadfcaf91bd98674ff8fc341d148e1731576a4

### CODE REVIEW

**Review by wpoulin — COMMENTED** [review]
Make sure to set return type within the function signature

File: src/apps/tai/ethereum/tai.collateral-fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @Inject(TaiContractFactory) protected readonly contractFactory: TaiContractFactory,
+  ) {
+  }
+
+  @Cache({
+    key: `studio:tai:collateral`,
+    ttl: 15 * 60,
+  })
+  async getCollateralTypes(network: Network) {
+    const { data } = await Axios.get<Record<string, string>>(`https://raw.githubusercontent.com/money-god/mgl-changelog/master/mainnet/contracts.json`);
```
[the remark this answers is not in the document]
Comment (tonzgao) [reply src/apps/tai/ethereum/tai.collateral-fetcher.ts:23]: asking here https://discord.com/channels/1019634528630878208/1114169602092564520/1116498080255184968, doesnt seem to have response though. there do appear to be some places where you can get a bytes32 name of collateral types, like here: https://etherscan.io/address/0x93b3968d50f465b94e19002f148cf66b103f13a2#readContract, but not sure how to get the address from there outside of the json list here

**Comment by tonzgao** [comment]
> hmm, so i dont want to look like im fighting you on this pr, but i dont think thats a good idea for maintenance reasons. is there a way to programmatically generate contract positions? if there isnt, then that would mean that each time a new collateral is added this pr would have to be manually updated. on the other hand, it would solve the issue of the hacky collateral fetch. open to changing it if you insist though

**Comment by wpoulin** [comment]
What I meant is something like this instead of having one definition which contains all the `collaterals`. 
Let me know if makes sense 

```
    return [
      collaterals.map(address => {
        return [
          { metaType: MetaType.SUPPLIED, address, network: this.network },
          { metaType: MetaType.BORROWED, address: '0xf915110898d9a455ad2da51bf49520b41655ccea', network: this.network }, // TAI
        ];
      }),
    ].flat();
  ```

**Comment by tonzgao** [comment]
> What I meant is something like this instead of having one definition which contains all the `collaterals`. Let me know if makes sense
> 
> ```
>     return [
>       collaterals.map(address => {
>         return [
>           { metaType: MetaType.SUPPLIED, address, network: this.network },
>           { metaType: MetaType.BORROWED, address: '0xf915110898d9a455ad2da51bf49520b41655ccea', network: this.network }, // TAI
>         ];
>       }),
>     ].flat();
> ```

discussed and fixed

**Review by wpoulin — APPROVED** [review]

**Comment by wpoulin** [comment]
You nailed it !

**Pull request merged**